ruma-federation-api/src/membership/create_leave_event/v1.rs

```diff
@@ -0,0 +1,86 @@
+//! [PUT /_matrix/federation/v1/send_leave/{roomId}/{eventId}](https://matrix.org/docs/spec/server_server/r0.1.4#put-matrix-federation-v1-send-leave-roomid-eventid)
+
+use std::time::SystemTime;
+
+use js_int::UInt;
+use ruma_api::ruma_api;
+use ruma_common::Raw;
+use ruma_events::{pdu::Pdu, room::member::MemberEventContent, EventType};
+use ruma_identifiers::{EventId, RoomId, ServerNameBox, UserId};
+
+ruma_api! {
+    metadata: {
+        description: "Submits a signed leave event to the resident server for it to accept it into the room's graph.",
+        name: "create_leave_event",
+        method: PUT,
+        path: "/_matrix/federation/v1/send_leave/:room_id/:event_id",
+        rate_limited: false,
+        requires_authentication: true,
+    }
+
+    #[non_exhaustive]
+    request: {
+        /// The room ID that is about to be left.
+        #[ruma_api(path)]
+        pub room_id: &'a RoomId,
+
+        /// The event ID for the leave event.
+        #[ruma_api(path)]
+        pub event_id: &'a EventId,
+
+        /// The user ID of the leaving member.
+        #[ruma_api(query)]
+        pub sender: &'a UserId,
+
+        /// The name of the leaving homeserver.
+        #[ruma_api(query)]
+        pub origin: ServerNameBox,
+
+        /// A timestamp added by the leaving homeserver.
+        #[ruma_api(query)]
+        #[serde(with = "ruma_serde::time::ms_since_unix_epoch")]
+        pub origin_server_ts: SystemTime,
+
+        /// The value `m.room.member`.
+        #[ruma_api(query)]
+        pub event_type: EventType,
```

Comment:
As-is, serde will expect a field / query param named `event_type`. You need to specify the differing name using an attribute:

```suggestion
        #[serde(rename = "type")]
        pub event_type: EventType,
```